src/wasm/wat-parser.cpp

```diff
@@ -573,10 +588,15 @@ struct ParseDeclsCtx : NullTypeParserCtx, NullInstrParserCtx {
   // The module element definitions we are parsing in this phase.
   std::vector<DefPos> typeDefs;
   std::vector<DefPos> subtypeDefs;
+  std::vector<DefPos> funcDefs;
   std::vector<DefPos> globalDefs;
 
+  // Positions of typeuses that might implicitly define new types.
```

Comment:
Maybe give an example here? I assume an example could be something like the implicit type that we give to functions with an explicit one?

Reply:
For example, block signatures are syntactically `typeuse`s, so if the [abbreviation](https://webassembly.github.io/spec/core/text/modules.html#abbreviations) is used, the block signature may implicitly define a new type. That type doesn't have a name, but it can be referred to by index, so it is important to get its position in the type index space correct. The current parser does not handle this correctly.

Reply:
Interesting.. I didn't know that was possible.